features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/telemetry/internal/TelemetryEventHandler.kt

```diff
@@ -74,38 +87,35 @@ internal class TelemetryEventHandler(
                         datadogContext = datadogContext,
                         timestamp = timestamp,
                         message = event.message,
-                        stack = event.stack,
+                        stack = event.stacktrace,
                         kind = event.kind,
                         additionalProperties = event.additionalProperties
                     )
                 }
 
-                TelemetryType.CONFIGURATION -> {
-                    val coreConfiguration = event.coreConfiguration
-                    if (coreConfiguration == null) {
-                        createErrorEvent(
-                            datadogContext = datadogContext,
-                            timestamp = timestamp,
-                            message = "Trying to send configuration event with null config",
-                            stack = null,
-                            kind = null,
-                            additionalProperties = null
-                        )
-                    } else {
-                        createConfigurationEvent(
-                            datadogContext,
-                            timestamp,
-                            coreConfiguration
-                        )
-                    }
+                is InternalTelemetryEvent.Configuration -> {
+                    createConfigurationEvent(
+                        datadogContext,
+                        timestamp,
+                        event
+                    )
                 }
 
-                TelemetryType.INTERCEPTOR_SETUP -> {
+                is InternalTelemetryEvent.ApiUsage -> {
+                    createApiUsageEvent(
+                        datadogContext = datadogContext,
+                        timestamp = timestamp,
+                        event = event
+                    )
+                }
+
+                is InternalTelemetryEvent.InterceptorInstantiated -> {
                     trackNetworkRequests = true
                     null
                 }
-            }
 
+                else -> null
```

Comment:
shouldn't we log here?

Reply:
actually that's a sealed class so we don't really need the `else` here anyway. In case we do not treat a case we will be warned by the editor.